### PR TITLE
feat: add violet config + exec env allowlist + helper functions (PR 1/3)

### PR DESCRIPTION
**Pull request by yhuan123**
## Summary

**PR 1 of 3** in the [violet externalization refactor](../blob/docs/violet-onboard-rolling-upgrade-plan/docs/plans/2026-05-18-refactor-externalize-violet-binary-plan.md). This PR is **purely additive** — no existing call site invokes the new code yet, so runtime behavior is unchanged.

The stack:
1. **PR 1 (this one)** — additive infra: config schema, exec env allowlist + stderr tail wrap, violet helper functions, table tests.
2. **PR 2** — swap \`InstallArtifactVersion\` to call \`violet push\`; keep the legacy function as dead code.
3. **PR 3** — delete the dead code.

## What changed

| File | Change |
|------|--------|
| \`pkg/exec/exec.go\` | Add \`Command.EnvAllowlist\` (exact / \`PREFIX_*\` glob). Empty allowlist preserves current full \`os.Environ()\` passthrough. Wrap last ~20 stderr lines into \`CommandResult.Err\` for actionable failures. |
| \`pkg/config/config.go\` | Add \`OperatorConfig.Violet *VioletConfig\` nested schema (\`Bin\` / \`PackagePrefix\` / \`SkipPush *bool\` / \`PushArgs\`) and \`Version.ExpectedSha256\`. \`defaultConfig\` fills MinIO prefix and \`SkipPush=true\` defaults when Violet is non-nil. \`*bool\` distinguishes \"unset\" from \"explicit false\". |
| \`pkg/operator/operatorhub/operator.go\` | Declare \`const targetCatalogSource = \"platform\"\` next to \`systemNamespace\`. |
| \`pkg/operator/operatorhub/violet.go\` (new) | Pure helpers: \`BuildPackageURL\`, \`BuildVioletPushArgs\` (auto-injects \`VIOLET_REGISTRY_USERNAME\`/\`_PASSWORD\` as \`--username\`/\`--password\`), \`MaskCommand\` (redacts password for logging), \`VerifySha256\`. |
| \`pkg/operator/operatorhub/violet_test.go\` (new) | Table tests: 22 sub-tests across all four helpers, covering both real channel formats (\`v4.6\` and \`rc\`-with-sha-suffix), skip-push variants, private-push flow, password masking edge cases, and SHA-256 verification (correct / uppercase / mismatch / missing file). |

## Testing

- \`go vet ./...\` — clean
- \`go build ./...\` — clean
- \`go test ./pkg/operator/operatorhub/... -run 'TestBuildPackageURL|TestBuildVioletPushArgs|TestMaskCommand|TestVerifySha256' -v\` — 22/22 pass

Nothing else in the codebase calls \`BuildPackageURL\` / \`BuildVioletPushArgs\` / \`MaskCommand\` / \`VerifySha256\` yet, so this PR cannot regress existing flows.

## Post-Deploy Monitoring & Validation

No additional operational monitoring required: this PR adds dormant code only and does not change any runtime call path. PR 2 introduces the actual behavior change and will carry the monitoring plan (download URL hit rate, violet exit codes, AV phase transition timing).

## Design references

- Brainstorm: \`docs/brainstorms/2026-05-18-violet-external-binary-brainstorm.md\` (on \`docs/violet-onboard-rolling-upgrade-plan\` branch)
- Plan: \`docs/plans/2026-05-18-refactor-externalize-violet-binary-plan.md\` (same branch)

### CODE REVIEW

**Review by kycheng — APPROVED** [review]

File: pkg/config/config.go [hunk 1]
```diff
@@ -71,8 +100,16 @@ type Version struct {
...
 	Channel string `yaml:"channel,omitempty"`
+	// ExpectedSha256, when non-empty, is the lowercase hex SHA-256 of the
+	// downloaded .tgz. The upgrade CLI verifies the digest after download and
+	// fails fast on mismatch. Optional; leave empty to skip verification.
+	ExpectedSha256 string `yaml:"expectedSha256,omitempty"`
 }
 
+// DefaultVioletPackagePrefix is the MinIO root used when OperatorConfig.Violet
+// is configured but its PackagePrefix is left blank.
+const DefaultVioletPackagePrefix = "http://package-minio.alauda.cn:9199/packages/"
```
Comment (kycheng) [inline pkg/config/config.go:111]: 不要硬编码在代码里
